enhancements/kata-containers/kata-containers-dev-preview.md

```diff
@@ -0,0 +1,259 @@
+
+**So What do we want to do with this?**
+
+We start by creating a container image with only the kata upstream components as static binaries (no RHEL-AV components). The kata operator will pull down this image and mount it in the host mount namespace, i.e. without running it as a container. We then use the RHCOS extensions for deploying the qemu-kiwi RHEL-AV RPMs.
+
+**Pros:**
+1. RHEL-AV depends on libraries and other tools which are all available as RPMs. Turning this into a static image instead of RPMs creates a huge single binary which we avoid with extensions
+2. RHEL-AV has been tested / validated as RPMs and not a single binary which means we saves the extra overhead
+3. RHEL-AV RPMs can be installed only by the customers who would be interested in using the kata runtime. QEMU (and its dependencies) get updated as part of the machine-os-content updates
+4. There's no extra work related to properly loading qemu-kiwi dependencies, as all the packages would be installed in the same PATH its build in
+5. Better control of the versions that are installed both for qemu (specific version for a given host) and for the dependencies (tracked by rpm)
+
+**Cons:**
+1. ~20 mb extra, installed in the host, for those who'd be using kata runtime
+2. machine-os-content would still carry this until there is a separate machine-os-content- extensions container
+
+This approach has a dependency on the extensions framework being delivered into Openshift.
+
+
+### KataContainer Operator development
+The goal is to develop a kata operator that can be used to manage the entire lifecycle of kata components in an OpenShift cluster. We will have a controller that watches for a kata **custom resource (CR)** and a daemon-set that acts upon changes to the (CR) and do the required work on the worker nodes (install, uninstall update,...). 
+
+Via the CR it will be possible to select a subset of worker nodes. For deploying binaries on the host the current idea is to use a container image that can be mounted on the host level.  The goal is to be able to install the operator via OperatorHub. The operator will also create a crio drop-in config file via machine config objects. The Operator will automatically select the payload image that contains correct version of the kata binaries for the given version of the OpenShift. The Operator will also configure a `RuntimeClass` called `kata` which can be used to deploy workload that uses kata runtime. The operator will create a machine-config object to enable the `qemu-kiwi` RHCOS extension.
+
+The `RuntimeClass` and payload image names will be visible in the CR status.
+
+#### Goals
+- Create an API which supports installation of Kata Runtime on all or selected worker nodes
+- Configure CRI-O to use Kata Runtime on those worker nodes
+- Installation of the runtimeClass on the cluster
+- Updates the Kata runtime
+- Uninstall Kata Runtime and reconfigure CRI-O to not use it.
```

Comment:
Add a reference to base scheduling metrics?

Reply:
@c3d Not sure what you'd like to see. Do you have one for me?

Reply:
I would add something like
`- Provide usable CPU, memory, disk and network metrics to the scheduler`

WDYT?

Reply:
@c3d, this is not necessarily the responsibility of Kata Containers, nor necessarily CRI-O or kubelet.  I'd argue against adding it, unless we have the full buy-in of the monitoring team to support those.

IMHO, adding this may actually increase the chance to have the addition blocked than anything else.

/cc @zanetworker

Reply:
I agree with you on this one @fidencio

Reply:
I agree with you both on the responsibilities. That is not my point however, nor is it the point of this section. The section describes the "Goals" of the operator, from the OpenShift perspective. I still believe that these goals should include adding whatever other configuration is needed for the orchestration layer to do its job. That includes metrics, pod overhead computations, and possibly other fine details that I'm sure we have not really covered yet.

What about extending the "Installation of the runtimeClass" bullet as follows:

```
- Installation of the runtimeClass on the cluster, as well as of the required components for the runtime to be controlled by the orchestration layer.
```

Come to think of it, this rewording is also necessary to describe the installation of the virtualization components.

Reply:
Made the change as suggested by @c3d